### PR TITLE
structopt: update magic_enum/0.9.5

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **structopt/***

magic_enum/0.9.5 changed include path from `"magic_enum.hpp"` to `magic_enum/magic_enum.hpp`.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — APPROVED** [review]

**Review by uilianries — CHANGES_REQUESTED** [review]
waiting for Ruben review

**Comment by uilianries** [comment]
I just pushed a PR to the upstream: https://github.com/p-ranav/structopt/pull/24

It should fix the current state by using patches.

**Comment by RubenRBS** [comment]
If the upstream PR does not get merged after this goes stale, we'll merge here as-is. The library seems not to be getting too many updates lately so they might take a bit

**Review by valgur — APPROVED** [review]

**Comment by toge** [comment]
It is not necessary modification.
Please refer https://github.com/conan-io/conan-center-index/pull/24894

**Pull request closed without merging**
